addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/timing/TimingUtils.java

```diff
@@ -83,9 +83,16 @@ public static boolean checkTimingDependence(
         int requestsLeft = requestsLimit;
         int requestsMade = 0;
         double secondsLeft = secondsLimit;
-        int currentDelay = MAX_DELAY - requestsLeft + 1;
+        // Respecting user configured delay limit (by default it is set to 15 seconds which is max
+        // delay & request limit often set to 5). Now, initial sleep duration will be (15-5+1) =
+        // 11 seconds, which get incremented by each second till 15 seconds
+        int currentDelay = maxDelay - requestsLeft + 1;
         double lastDelay = 0;
 
+        if (currentDelay < 0) {
```

Comment:
`<=` ?

Would we ever want it to actually be zero?

Reply:
this would work fine. But, i agree in actual we don't need it to zero